### PR TITLE
chore(Auth): migrate auth operations to async AmplifyAuthTasks

### PR DESCRIPTION
**Pull request by phantumcode**
*Issue #, if available:*

*Description of changes:*
Migrate the rest of auth operations and public apis to async/await pattern
*Check points: (check or cross out if not relevant)*

- [x] Added new tests to cover change, if needed
- [x] Build succeeds with all target using Swift Package Manager
- [x] All unit tests pass
- [ ] ~All integration tests pass~
- [x] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [x] Documentation update for the change if required
- [x] PR title conforms to conventional commit style
- [x] If breaking change, documentation/changelog update with migration instructions

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Task/AWSAuthFetchSessionTask.swift [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                    let session = try result.get()
+                    continuation.resume(returning: session)
+                } catch {
+                    continuation.resume(throwing: error)
+                }
+            }
+        }
+    }
+    
+    private func didConfigure() async {
```
Comment (royjit) [inline AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Task/AWSAuthFetchSessionTask.swift:42]: We could move this to an helper function to be reused in all tasks.

**Review by royjit — DISMISSED** [review]
LGTM, minor comments

File: AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Task/AWSAuthFetchSessionTask.swift (hunk 1 above)
Comment (phantumcode) [reply AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Task/AWSAuthFetchSessionTask.swift]: Done, moved to a helper class.

File: AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Task/UserTasks/AWSAuthFetchUserAttributeTask.swift [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        await withCheckedContinuation { [weak self] (continuation: CheckedContinuation<Void, Never>) in
+            stateMachineToken = authStateMachine.listen({ [weak self] state in
+                guard let self = self, case .configured = state else { return }
+                self.authStateMachine.cancel(listenerToken: self.stateMachineToken!)
+                continuation.resume()
+            }, onSubscribe: {})
+        }
+    }
+
+    private func getAccessToken() async throws -> String {
```
[the remark this answers is not in the document]
Comment (phantumcode) [reply AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Task/UserTasks/AWSAuthFetchUserAttributeTask.swift]: Done, moved to a helper class.

**Review by royjit — APPROVED** [review]

**Pull request merged**